frontend/src/employee-frontend/components/messages/MessageList.tsx

```diff
@@ -43,6 +44,27 @@ export default React.memo(function MessagesList({
     [account.id, selectThread, view]
   )
 
+  const [messageCount, setMessageCount] = useState<number>(0)
+
+  useEffect(() => {
+    if (view === 'RECEIVED' && receivedMessages.isSuccess) {
+      setMessageCount(receivedMessages.value.length)
+    } else if (view === 'SENT' && sentMessages.isSuccess) {
+      setMessageCount(sentMessages.value.length)
+    } else if (view === 'DRAFTS' && messageDrafts.isSuccess) {
```

Comment:
Wouldn't the `.isSuccess` states here mean that no "in progress" state is shown to the user if loading messages takes longer than rendering this once?

Reply:
Added an "in progress" view